internal/ngap/handler_generated.go

```diff
@@ -179,17 +212,65 @@ func handlerAMFConfigurationUpdateAcknowledge(ran *context.AmfRan, successfulOut
 	for _, ie := range aMFConfigurationUpdateAcknowledge.ProtocolIEs.List {
 		switch ie.Id.Value {
 		case ngapType.ProtocolIEIDAMFTNLAssociationSetupList: // optional, ignore
-			ran.Log.Info("Not comprehended IE AMF-TNLAssociationSetupList")
+			if aMFTNLAssociationSetupList != nil {
+				ran.Log.Error("Duplicate IE AMF-TNLAssociationSetupList")
+				abort = true
+				break
+			}
+			aMFTNLAssociationSetupList = ie.Value.AMFTNLAssociationSetupList
+			ran.Log.Trace("Decode IE AMF-TNLAssociationSetupList")
 		case ngapType.ProtocolIEIDAMFTNLAssociationFailedToSetupList: // optional, ignore
-			ran.Log.Info("Not comprehended IE TNLAssociationList")
+			if aMFTNLAssociationFailedToSetupList != nil {
+				ran.Log.Error("Duplicate IE TNLAssociationList")
+				abort = true
+				break
+			}
+			aMFTNLAssociationFailedToSetupList = ie.Value.AMFTNLAssociationFailedToSetupList
+			ran.Log.Trace("Decode IE TNLAssociationList")
 		case ngapType.ProtocolIEIDCriticalityDiagnostics: // optional, ignore
 			if criticalityDiagnostics != nil {
 				ran.Log.Error("Duplicate IE CriticalityDiagnostics")
-				return
+				abort = true
+				break
 			}
 			criticalityDiagnostics = ie.Value.CriticalityDiagnostics
 			ran.Log.Trace("Decode IE CriticalityDiagnostics")
+		default:
+			switch ie.Criticality.Value {
+			case ngapType.CriticalityPresentReject:
+				ran.Log.Errorf("Not comprehended IE ID 0x%04x (criticality: reject)", ie.Id.Value)
+			case ngapType.CriticalityPresentIgnore:
+				ran.Log.Infof("Not comprehended IE ID 0x%04x (criticality: ignore)", ie.Id.Value)
+			case ngapType.CriticalityPresentNotify:
+				ran.Log.Warnf("Not comprehended IE ID 0x%04x (criticality: notify)", ie.Id.Value)
+				item := buildCriticalityDiagnosticsIEItem(ie.Criticality.Value, ie.Id.Value, ngapType.TypeOfErrorPresentNotUnderstood)
+				iesCriticalityDiagnostics.List = append(iesCriticalityDiagnostics.List, item)
+			}
+		}
+	}
+
+	if syntaxCause != nil || len(iesCriticalityDiagnostics.List) > 0 {
+		ran.Log.Trace("Has IE error")
+		procedureCode := ngapType.ProcedureCodeAMFConfigurationUpdate
+		triggeringMessage := ngapType.TriggeringMessagePresentSuccessfulOutcome
+		procedureCriticality := ngapType.CriticalityPresentReject
+		var pIesCriticalityDiagnostics *ngapType.CriticalityDiagnosticsIEList
+		if len(iesCriticalityDiagnostics.List) > 0 {
+			pIesCriticalityDiagnostics = &iesCriticalityDiagnostics
 		}
+		criticalityDiagnostics := buildCriticalityDiagnostics(&procedureCode, &triggeringMessage, &procedureCriticality, pIesCriticalityDiagnostics)
+		ngap_message.SendErrorIndication(ran, nil, nil, syntaxCause, &criticalityDiagnostics)
+	}
+
+	if abort {
+		return
+	}
+
+	if aMFTNLAssociationSetupList != nil {
+		ran.Log.Info("IE AMF-TNLAssociationSetupList is not implemented")
```

Comment:
Will it be better to use a Warn log here?

Reply:
I changed log level.